src/Internal/Types/BigInt.purs

```diff
@@ -0,0 +1,150 @@
+-- | Arbitrary-precision integers (backed by 'purescript-js-bigint').
+module Ctl.Internal.Types.BigInt
+  ( BigInt
+  , toString
+  , toNonEmptyString
+  , fromString
+  , fromBase
+  , toBase
+  , toBase'
+  , toInt
+  , fromInt
+  , fromTLInt
+  , toNumber
+  , fromNumber
+  , abs
+  , even
+  , odd
+  , pow
+  ) where
+
+import Prelude
+
+import Aeson
+  ( class DecodeAeson
+  , class EncodeAeson
+  , JsonDecodeError(TypeMismatch)
+  , decodeAeson
+  , encodeAeson
+  )
+import Data.Either (Either(Left))
+import Data.Int (decimal, radix)
+import Data.Maybe (Maybe(Nothing, Just))
+import Data.Maybe (fromMaybe)
+import Data.Reflectable (class Reflectable)
+import Data.String.NonEmpty.Internal (NonEmptyString(NonEmptyString))
+import JS.BigInt as BigIntImpl
+import Prim (Boolean, Int, Number, String, Symbol)
+import Prim.Int (class ToString)
+import Type.Proxy (Proxy)
+
+newtype BigInt = UnsafeBigInt BigIntImpl.BigInt
```

Comment:
Is there any other reason other than Aeson to have a newtype? If not, please just make `aeson` package depend on `js-bigints` (and "undepend" it from `bigints`) and put the instances near the aeson typeclasses. Then remove this newtype wrapper. The reason is: we want to make it easier to integrate with the rest of the purescript ecosystem libraries - they know about js-bigints, but working with our BigInt would require (un)wrapping and will lead to more confusion.

Reply:
That was the only reason really. I'll send a PR to `aeson`.